From 5e4db7bc5735f7e635d8ecd1af480bced4ae3087 Mon Sep 17 00:00:00 2001
From: Debbie Martin <Debbie.Martin@arm.com>
Date: Tue, 5 Sep 2023 21:37:46 +0100
Subject: [PATCH] check-sr-results: Return non-zero exit code on failure

Return a non-zero exit code if there are any warnings or errors found.

Upstream-Status: Pending
Signed-off-by: Debbie Martin <Debbie.Martin@arm.com>

%% original patch: 0001-check-sr-results-Return-non-zero-exit-code-on-failur.patch
---
 check-sr-results.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/check-sr-results.py b/check-sr-results.py
index ebc523b..3517412 100755
--- a/check-sr-results.py
+++ b/check-sr-results.py
@@ -202,6 +202,9 @@ class Stats:
     def inc_error(self):
         self._inc('error')
 
+    def has_warnings_or_errors(self):
+        return self.data['warning'] or self.data['error']
+
 
 # Download (possibly large) file from URL.
 # We raise an exception in case of issue.
@@ -1679,3 +1682,6 @@ if __name__ == '__main__':
     if args.print_meta:
         print()
         print_meta()
+
+    if stats.has_warnings_or_errors():
+        exit(1)
-- 
2.25.1

